xdsl/utils/mlir_lexer.py

```diff
@@ -250,6 +223,30 @@ def get_string_literal_value(self, span: Span) -> str:
         return StringLiteral.from_span(span).string_contents
 
 
+KIND_BY_PUNCTUATION_SPELLING = {
```

Comment:
Worth considering having this as a ClassVar on `MLIRTokenKind`?

Reply:
It's not happy when I try to do this directly, as the enums are not fully initialised until the end of the class definition, and doing some kind of setting a class var after the class definition also seems a little off.